ql/src/test/org/apache/hadoop/hive/ql/TestTxnExIm.java

```diff
@@ -566,4 +602,44 @@ public void testMMExportAborted() throws Exception {
         TestTxnCommands2.stringifyValues(data), rs);
 
   }
-}
+
+  @Test
+  public void testImportOrc() throws Exception {
+    //Clear and Drop T and Tstage if exist
+    runStatementOnDriver("drop table if exists T");
+    runStatementOnDriver("drop table if exists Tstage");
+
+    //create target table - T
+    runStatementOnDriver("create table T (a int, b int) stored" + " as orc tblproperties('transactional'='true')");
+
+    //Create source table - Tstage
+    runStatementOnDriver("create table Tstage (a int, b int) stored" + " as orc tblproperties('transactional'='true')");
+
+    //this creates an ORC data file with correct schema under table root
+    runStatementOnDriver("insert into Tstage values(1,2),(3,4),(5,6)");
+    final int[][] rows = { { 3 } };
+
+    //check Tstage statistics
+    List<String> rsTStageProperties = runStatementOnDriver("show tblproperties Tstage");
+    Assert.assertEquals("COLUMN_STATS_ACCURATE of Tstage table", true,
+        rsTStageProperties.contains("COLUMN_STATS_ACCURATE\t{\"BASIC_STATS\":\"true\"}"));
+    Assert.assertEquals("numRows of Tstage table", true, rsTStageProperties.contains("numRows\t3"));
+    Assert.assertEquals("numFiles of Tstage table", true, rsTStageProperties.contains("numFiles\t1"));
+
+    //now we have an archive Tstage table
+    runStatementOnDriver("export table Tstage to '" + getWarehouseDir() + "/1'");
+
+    //load T
+    runStatementOnDriver("import table T from '" + getWarehouseDir() + "/1'");
+
+    //check basic stats in tblproperties T
+    List<String> rsTProperties = runStatementOnDriver("show tblproperties T");
```

Comment:
Can test the import with hive.stats.autogather=true and see if stats are accurate for both partitioned and non-partitioned tables.